### PR TITLE
Delete partially fetched packages on failure.

### PR DESCRIPTION
**Pull request by etefera**
Prevents cluttering the destination space when a `pkg get` fails, addressing https://github.com/GoogleContainerTools/kpt/issues/2323

### CODE REVIEW

File: internal/util/get/get.go [hunk 1]
```diff
@@ -84,18 +86,30 @@ func (c Command) Run(ctx context.Context) error {
...
 	}
+	p.DisplayPath = types.DisplayPath(filepath.Base(c.Destination))
 
 	if err = c.fetchPackages(ctx, p); err != nil {
-		return errors.E(op, types.UniquePath(c.Destination), err)
+		return errors.E(op, types.UniquePath(tempPath), err)
+	}
+
+	// Move temporary directory into final destination.
+	err = os.MkdirAll(filepath.Dir(c.Destination), 0700)
```
Comment (phanimarupaka) [inline internal/util/get/get.go:105]: Have you explored clean-up on error option as opposed to writing to temp and copying over? If get behavior is dependent on parent package state then this approach will not work. Previously, setter inheritance used to depend on package structure so this approach wouldn't work. But I feel it is better to implement clean up on failure. There might be features in the future where get might depend on parent packages.

@mortent

Comment (mortent) [reply internal/util/get/get.go:105]: I also think cleaning up on failure would be a better option here, unless there is a reason why that wouldn't work.

Comment (etefera) [reply internal/util/get/get.go]: Done.

File: internal/util/get/get.go [hunk 2]
```diff
@@ -95,6 +95,10 @@ func (c Command) Run(ctx context.Context) error {
 	}
 
 	if err = c.fetchPackages(ctx, p); err != nil {
+		rmErr := os.RemoveAll(c.Destination)
```
Comment (phanimarupaka) [inline internal/util/get/get.go:98]: Clean up should happen for any error after line 68.

**Review by phanimarupaka — CHANGES_REQUESTED** [review]

File: internal/util/get/get.go (hunk 2 above)
Comment (etefera) [reply internal/util/get/get.go]: Done.

**Review by phanimarupaka — APPROVED** [review]

**Pull request merged**